adapters/adorsys-adapter/src/main/java/de/adorsys/xs2a/adapter/adorsys/AdorsysIntegServiceProvider.java

```diff
@@ -17,63 +17,45 @@
 package de.adorsys.xs2a.adapter.adorsys;
 
 import de.adorsys.xs2a.adapter.api.*;
-import de.adorsys.xs2a.adapter.api.config.AdapterConfig;
 import de.adorsys.xs2a.adapter.api.http.HttpClient;
 import de.adorsys.xs2a.adapter.api.http.HttpClientFactory;
-import de.adorsys.xs2a.adapter.api.http.Interceptor;
 import de.adorsys.xs2a.adapter.api.link.LinksRewriter;
 import de.adorsys.xs2a.adapter.api.model.Aspsp;
 import de.adorsys.xs2a.adapter.impl.BasePaymentInitiationService;
 import de.adorsys.xs2a.adapter.impl.http.RequestSigningInterceptor;
 import de.adorsys.xs2a.adapter.impl.oauth2.api.BaseOauth2Api;
 import de.adorsys.xs2a.adapter.impl.oauth2.api.model.AuthorisationServerMetaData;
 
-import java.time.ZonedDateTime;
-import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
 
 public class AdorsysIntegServiceProvider
     implements AccountInformationServiceProvider, PaymentInitiationServiceProvider, Oauth2ServiceProvider {
 
     private final OauthHeaderInterceptor oauthHeaderInterceptor = new OauthHeaderInterceptor();
-    private final boolean requestSigningEnabled =
-        Boolean.parseBoolean(AdapterConfig.readProperty("adorsys.request_signing.enabled", "false"));
+    private AdorsysSigningHeadersInterceptor adorsysSigningHeadersInterceptor;
 
     @Override
     public PaymentInitiationService getPaymentInitiationService(Aspsp aspsp,
                                                                 HttpClientFactory httpClientFactory,
                                                                 Pkcs12KeyStore keyStore,
                                                                 LinksRewriter linksRewriter) {
+        adorsysSigningHeadersInterceptor = new AdorsysSigningHeadersInterceptor(new RequestSigningInterceptor(keyStore));
         return new BasePaymentInitiationService(aspsp,
                                                 httpClientFactory.getHttpClient(getAdapterId()),
-                                                keyStore != null ? getInterceptor(keyStore) : null,
+                                                Arrays.asList(oauthHeaderInterceptor, adorsysSigningHeadersInterceptor),
```

Comment:
and what about the case when a keystore is null?

Reply:
As I understood, this null checking was here only to make WireMock tests running. I tweaked the test a bit to negate this checking.
Can it ever be null?
It's not possible to start the Adapter app without a KeyStore. Besides, this is the only place in Adapter where KeyStore is checked for null.

Reply:
done